deducing-this.md

```diff
@@ -366,15 +391,20 @@ It also works as a more obvious way to teach how `std::bind` and `std::function`
 
 ## ABI implications for `std::function` and related
 
-If references and pointers do not have the same representation for member functions, this effectively says "for the purposes of the `this`-designated first parameter, they do."
+The representation of references is not defined. This leads to the following
+constraint: if references and pointers do not have the same representation for member functions, this effectively says "for the purposes of the `this`-designated first parameter, they do."
 
 This matters because code written in the "`this` is a pointer" syntax with the `this->` notation needs to be assembly-identical to code written with the `self.` notation; the two are just different ways to implement a function with the same signature.
 
-## Interplays with capturing `[this]` and `[*this]`
+Note that since `this` is a pointer, one cannot overoad `operator->` for it, and
+therefore `this->` and `self.` are always equivalent expressions (at least until
+we get `operator.`)
+
+## Interplays with capturing `[this]` and `[*this]` in lambdas
 
 `this` just designates the parameter that is bound to the reference to the function object. It does not, in any way, change the meaning of `this`.
 
-If other language features play with what `this` means, they are completely orthogonal and do not have interplays with this proposal. However, it should be obvious that develpers have a great potential for introducing hard-to-read code if they are at all changing the meaning of `this`, especially in conjunction with this proposal.
+If other language features play with what `this` inside a function body means, they are completely orthogonal, and do not have interplays with this proposal, since this paper proposes no change to that behavior. However, it should be obvious that develpers already have a great potential for introducing hard-to-read code if they are at all changing the meaning of `this`, especially in conjunction with this proposal.
```

Comment:
I just wanted to leave a comment that I love the phrase:

> it should be obvious that developers already have a great potential for introducing hard-to-read-code

The "a" is extra though, just "... already have great potential for ..."

Reply:
Thanks! Fixed.